soc/xtensa/esp32/linker.ld

```diff
@@ -19,28 +19,31 @@
 
 #define RAMABLE_REGION dram0_0_seg :dram0_0_phdr
 #define ROMABLE_REGION iram0_0_seg :iram0_0_phdr
+#define FLASH_CODE_REGION irom0_0_seg :irom0_0_phdr
+#define FLASH_DATA_REGION drom0_0_seg :drom0_0_phdr
 
 PROVIDE ( __stack = 0x3ffe3f20 );
 
 PROVIDE ( esp32_rom_uart_tx_one_char = 0x40009200 );
 PROVIDE ( esp32_rom_uart_rx_one_char = 0x400092d0 );
 PROVIDE ( esp32_rom_uart_attach = 0x40008fd0 );
 PROVIDE ( esp32_rom_uart_tx_wait_idle = 0x40009278 );
-PROVIDE ( esp32_rom_intr_matrix_set = 0x4000681c );
 PROVIDE ( esp32_rom_gpio_matrix_in = 0x40009edc );
 PROVIDE ( esp32_rom_gpio_matrix_out = 0x40009f0c );
 PROVIDE ( esp32_rom_Cache_Flush = 0x40009a14 );
 PROVIDE ( esp32_rom_Cache_Read_Enable = 0x40009a84 );
 PROVIDE ( esp32_rom_ets_set_appcpu_boot_addr = 0x4000689c );
 PROVIDE ( esp32_rom_i2c_readReg = 0x40004148 );
 PROVIDE ( esp32_rom_i2c_writeReg = 0x400041a4 );
+PROVIDE ( ets_printf = 0x40007d54 );
```

Comment:
Since this is provide by the ROM, having a prefix of `esp32_rom_` would make it consistent with other functions here.

Reply:
@dcpleung, yes, it makes sense. However, to make it easier for future esp-idf rebase and updates, we woud like to maintain consistence. Soon we will update this.

Reply:
Please fix it before this PR is merged.  The name exposed by the Zephyr linker script should not need to be the same as the name used in the ROM, nor the name used in `esp-idf`.

One way to make this cleaner is to add the same function with two aliases:
* one using the spelling used in `esp-idef` until it's cleaned up;
* another with the `esp32_rom_` prefix used in Zephyr

Use in Zephyr should use the name with the proper prefix, and the secondary alias should have a comment explaining why it's there and the conditions to be met so it can be removed.  All those secondary aliases could be grouped together to help make that clear.